### ISSUE TITLE
Sprite texture becomes dark when using a shared texture with a MeshInstance's surface material.

### ISSUE THREAD

**Issue by Teknologicus** [godotengine/godot#18438]
**Godot version:**
3.0.2 (stable)

**OS/device including version:**
Kubuntu Linux 16.0.4 / Nvidia GeForce GTX 560

**Issue description:**
If the same texture is used on a Sprite (2d scene) and on a MeshInstance's surface material (3d scene) at the same time, sprite's texture is darkened noticeably.

**Steps to reproduce:**
Create a Sprite node and a MeshInstance (CubeMesh).  Set both the Sprite's texture and MeshInstance's surface material to the same texture.  Run project with MeshInstance hidden and then shown to see how the Sprite is dark when the MeshInstance is visible.

**Minimal reproduction project:**
[sprite darkness bug.zip](https://github.com/godotengine/godot/files/1950784/sprite.darkness.bug.zip)

**Comment by Teknologicus** [comment]
P.S.  The work around is to make a copy of the texture file and use one copy only for Sprite(s) and the other only for 3D scene nodes.

**Comment by SaracenOne** [comment]
@Teknologicus I actually encountered this recently when messing about with some 3D viewports, and I discovered that it's all related sRGB colour conversion, specifically the way in which the 3D renderer expects sRGB color space for textures as opposed to the canvas renderer, and will attempt to convert to sRGB if it appears in the 3D renderer. There are a couple of ways around this, one as you already listed of having multiple version of the textures, but there's also a SpatialMaterial flag which handles colour conversion at the shader stage.

It's debatable whether this is 'actually' a bug or not; it's certainly quite cryptic, especially since most people probably don't really think about colour space too much, but one thing which might be a bug is how the automated sRGB conversion doesn't seem to allowed to be disabled, and doesn't seem to respect import settings.

Not sure what the best call to make here would be.
